src/viewer.js

```diff
@@ -2919,7 +2881,30 @@ function onCanvasKeyPress( event ) {
     }
 }
 
+function onCanvasKeyPress( event ) {
+    var canvasKeyPressEventArgs = {
+      originalEvent: event.originalEvent,
+      preventDefaultAction: false,
```

Comment:
This property is meaningless, because there's no default action in this function... It just raises the event and doesn't do anything else. Same with the next two properties (preventVerticalPan and preventHorizontalPan). Please remove them.

Reply:
@iangilman while I agree with this, I just can't shake the feeling that it's going to bite us when the viewer lets preventDefault through as false on the deprecated keypress events. It should be rare that the viewer is on top of an element that has default actions but I always seem to get bit by stuff like this. 😏 Anyway I've thought it through and I'm ok with it

Reply:
@msalsbery I think I understand what you're referring to, but I don't think it's the case here. We don't translate `preventDefaultAction` to the original MouseTracker event; it's only used here in this handler. Should we be translating that through? If we do, it would be a new behavior. 

And remember, the `canvas-key-press` event is new since last release, so there's no legacy code that uses it. The event that has existed is `canvas-key` and it doesn't translate the `preventDefaultAction` into the MouseTracker event. 

Does that address your concerns, or am I missing something?

Reply:
@iangilman  preventDefaultAction can be removed, absolutely. What I'm referring to is the event.preventDefault flag which is set for any key presses the viewer handles.  The default is false. It should be set true to prevent the user agent's (or another MouseTracker or element above that honors it) default action. Since the original issue was double events, we're now passing preventDefault true in keydown handler and false in keypress handler for the keys the viewer handles.

That being said, I can't think of any default key handling behavior that would be under a viewer element. If there was, we would have received a filed issue for it by now I think. That's why I'm ok with it ;)

Reply:
Ship it! :shipit:

Reply:
Great, makes sense! I think we're good. :)